src/vip_ipykernel_launcher.py

```diff
@@ -0,0 +1,26 @@
+"""Taken and modified from
+https://github.com/ipython/ipykernel/blob/master/ipykernel_launcher.py
+
+Entry point for launching an ViP-IPython kernel.
+"""
+
+import subprocess
+import sys
+
+from vip_ipykernel import venv_search
+
+if __name__ == '__main__':
+    # Remove the CWD from sys.path while we load stuff.
+    # This is added back by InteractiveShellApp.init_path()
+    if sys.path[0] == '':
+        del sys.path[0]
+
+    args = sys.argv
+
+    args[0] = str(venv_search())
+
+    #  TODO: I want to use the jupyter logger to print this off but can't figure
+    #  out how to do that, @takluyver can you help with this?
+    print(f"Starting venv kernel with args: {args}")
```

Comment:
I don't think anything has set up logging - you haven't even imported any Jupyter/IPython modules at this point, as far as I can see. You could call e.g. `logging.basicConfig()` yourself, but I don't think there's any major advantage over plain `print()`.

Reply:
Basically I just wanted, for no real reason, to print this output to the Jupyter Notebook/Lab logging. Currently it looks like:

```
[I 13:54:14.648 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 13:54:14.709 NotebookApp]

    To access the notebook, open this file in a browser:
        file:///home/roscar/.local/share/jupyter/runtime/nbserver-3635158-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=5e4191cd8032e03cf7618509400265a9e6d67e56b98f1eb3
     or http://127.0.0.1:8888/?token=5e4191cd8032e03cf7618509400265a9e6d67e56b98f1eb3
[I 13:54:56.902 NotebookApp] Kernel started: 1fc13264-3b04-4015-85b9-6452750217ae, name: python3
Starting venv kernel with args: ['/usr/bin/python3.8', '-m', 'ipykernel_launcher', '-f', '/home/roscar/.local/share/jupyter/runtime/kernel-1fc13264-3b04-4015-85b9-6452750217ae.json']
```

And I'd like:

```
[I 13:54:14.648 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 13:54:14.709 NotebookApp]

    To access the notebook, open this file in a browser:
        file:///home/roscar/.local/share/jupyter/runtime/nbserver-3635158-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=5e4191cd8032e03cf7618509400265a9e6d67e56b98f1eb3
     or http://127.0.0.1:8888/?token=5e4191cd8032e03cf7618509400265a9e6d67e56b98f1eb3
[I 13:54:56.902 NotebookApp] Kernel started: 1fc13264-3b04-4015-85b9-6452750217ae, name: python3
[I 13:54:SS.sss NotebookApp]  Starting venv kernel with args: ['/usr/bin/python3.8', '-m', 'ipykernel_launcher', '-f', '/home/roscar/.local/share/jupyter/runtime/kernel-1fc13264-3b04-4015-85b9-6452750217ae.json']
```

I've tried using `logging.basicConfig()` to get the current logger, but that doesn't pick up the one the server is using, also tried figuring out how to access the logger instance traitlet (https://github.com/ipython/ipykernel/blob/master/ipykernel/kernelbase.py#L68) but I've never really used or looked into traitlets so they're still pretty mysterious to me

tbh you're right since `print` achieves the same thing, all that's missing is the timestamp so if this is too much effort then that's fine since it really doesn't matter much, I was just curious if there's an easy way to use the logger directly

Reply:
Your code is running in a child process of the notebook server, so there's no way to access the server's logging machinery directly, and AFAIK Jupyter doesn't expose any special interfaces for kernels to do logging. If you want the timestamp, you could set up the logging framework in your kernel process, but I probably wouldn't bother if it was me.

Reply:
Yeah that's what I thought, print is good enough for this so I'll leave it as it is